### PR TITLE
feat(flags): add semver targeting to local evaluation

### PR DESCRIPTION
**Pull request by dmarticus**
## Problem

Users cannot locally evaluate feature flags that use semver (semantic versioning) property filters. When a flag uses operators like `semver_gt` or `semver_caret`, the Node SDK must fall back to server-side evaluation, adding latency and requiring network calls.

This brings the posthog-node SDK in line with:
- PostHog server: PostHog/posthog#44596
- Python SDK: PostHog/posthog-python#447

## Changes

- Add 9 semver operators to local feature flag evaluation in `matchProperty`:
  - `semver_eq`, `semver_neq` — equality/inequality comparisons
  - `semver_gt`, `semver_gte`, `semver_lt`, `semver_lte` — standard comparisons
  - `semver_tilde` — patch-level range (~1.2.3 means >=1.2.3 and <1.3.0)
  - `semver_caret` — compatible-with range per semver spec (handles 0.x specially)
  - `semver_wildcard` — wildcard range (1.2.* means >=1.2.0 <1.3.0)
- Add `parseSemver` function that:
  - Parses version strings into [major, minor, patch] tuples
  - Strips v-prefix, whitespace, pre-release, and build metadata
  - Defaults missing components to 0
  - Throws `InconclusiveMatchError` for invalid input
- Add helper functions for range computations: `computeTildeBounds`, `computeCaretBounds`, `computeWildcardBounds`
- Export `parseSemver` for testing
- Add comprehensive test coverage (177 tests) covering all operators, edge cases, and error handling

**Implementation notes:**
- No external semver library used (inline implementation per SDK guidelines)
- Pre-release and build metadata are stripped before comparison (e.g., `1.2.3-alpha` == `1.2.3`)
- Follows the same semantics as the Python SDK implementation

### Libraries affected

- [ ] All of them
- [ ] posthog-js (web)
- [ ] posthog-js-lite (web lite)
- [x] posthog-node
- [ ] posthog-react-native
- [ ] @posthog/react
- [ ] @posthog/ai
- [ ] @posthog/convex
- [ ] @posthog/nextjs-config
- [ ] @posthog/nuxt
- [ ] @posthog/rollup-plugin
- [ ] @posthog/webpack-plugin
- [ ] @posthog/types

## Checklist

- [x] Tests for new code
- [x] Accounted for the impact of any changes across different platforms
- [x] Accounted for backwards compatibility of any changes (no breaking changes!)
- [x] Took care not to unnecessarily increase the bundle size

### If releasing new changes

- [ ] Ran `pnpm changeset` to generate a changeset file
- [ ] Added the "release" label to the PR to indicate we are publishing new versions for the affected packages

### CODE REVIEW

File: packages/node/src/extensions/feature-flags/feature-flags.ts [hunk 1]
```diff
@@ -1082,6 +1082,48 @@ function matchProperty(
...
+          return cmp > 0
+        case 'semver_gte':
+          return cmp >= 0
+        case 'semver_lt':
+          return cmp < 0
+        case 'semver_lte':
+          return cmp <= 0
+        default:
+          throw new InconclusiveMatchError(`Unknown semver operator: ${operator}`)
+      }
```
[the remark this answers is not in the document]
Comment (gustavohstrassburger) [reply packages/node/src/extensions/feature-flags/feature-flags.ts]: This is legit.

File: packages/node/src/__tests__/feature-flags.spec.ts
Comment (gustavohstrassburger) [inline packages/node/src/__tests__/feature-flags.spec.ts:1]: Nice work on the tests.

**Comment by gustavohstrassburger** [comment]
Don't forget the release label.
